app/models/miq_queue_worker_base/runner.rb

```diff
@@ -195,13 +195,25 @@ def miq_messaging_listener_thread
   def miq_messaging_subscribe_topic(&block)
     messaging_client = MiqQueue.messaging_client(self.class.name)
     messaging_client.subscribe_topic(:service => "manageiq.#{@worker.queue_name}", :persist_ref => @worker.guid, &block)
+  rescue Rdkafka::RdkafkaError => err
```

Comment:
I wonder if we should be returning ManageIQ::Messaging errors instead?  Not sure.

Reply:
Yeah agreed don't like when gems raise exceptions from their internal implementations